jwt/transport/transport.go

```diff
@@ -0,0 +1,175 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// ErrTokenRefresh is an error type returned by `RoundTrip` if the bouncer response was not 200.
+type ErrTokenRefresh struct {
+	msg string
+}
+
+func (e ErrTokenRefresh) Error() string {
+	return e.msg
+}
+
+// ImplWithJWT is a wrapper over http.RoundTripper which adds a valid token to each request.
+type ImplWithJWT struct {
```

Comment:
still dislike that this is a public type. the only public members are Lock and Unlock (which probably don't make sense to expose publically)